### PR TITLE
Test foreign strings on invalid content more thoroughly

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

File: validation-test/stdlib/StringUTF8.swift [hunk 1]
```diff
@@ -47,8 +48,33 @@ var strings: Array<String> = [
   "日",
 ]
 
+var utf16ByteSequences = utf16Tests.flatMap { $0.value }.map { $0.encoded }
+
 let kCFStringEncodingASCII: UInt32 = 0x0600
 
+private func testForeignContiguous(slowString: NSSlowString, string: String) {
```
Comment (Catfish-Man) [inline validation-test/stdlib/StringUTF8.swift]: This is just the bit below pulled out so I can reuse it for the second set of data

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
"LLDB :: SwiftREPL/BreakpointSimple.test" is absolutely not caused by adding unrelated tests

**Comment by Catfish-Man** [comment]
@swift-ci please test osx platform

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
This time Interpreter/switch_resilience.swift :/ still not caused by this

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
Test run hung >.<

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test linux platform

File: validation-test/stdlib/StringUTF8.swift [hunk 2]
```diff
@@ -49,6 +50,35 @@ var strings: Array<String> = [
...
+
+private func testForeignContiguous(slowString: NSSlowString, string: String) {
+  // Lazily bridged strings are not contiguous UTF-8
+  var slowString = NSSlowString(string: string) as String
+  expectFalse(slowString.isFastUTF8)
+  expectEqualSequence(string.utf8, slowString.utf8)
+  
+  // They become fast when mutated
+  slowString.makeNative()
+  expectTrue(slowString.isFastUTF8)
```
Comment (milseman) [inline validation-test/stdlib/StringUTF8.swift]: Oh right, this test predates contiguous strings proposal. You can go this route, or update the test to use `isContiguousUTF8`, `makeContiguousUTF8()`, and `withUTF8()`.

**Review by milseman — APPROVED** [review]
LGTM

File: validation-test/stdlib/StringUTF8.swift (hunk 2 above)
Comment (Catfish-Man) [reply validation-test/stdlib/StringUTF8.swift]: Noted. I'm probably going to leave it as unchanged as possible for now.

**Pull request merged**